APCAppCore/APCAppCore/Library/APCTaskImporter.m

```diff
@@ -0,0 +1,489 @@
+                                        nestedError: errorLoadingSurveyFile];
+            
+            APCLogError2 (error);
+        }
+        
+        else
+        {
+            NSError *errorParsingSurveyContent = nil;
+            NSDictionary *surveyContent = [NSJSONSerialization JSONObjectWithData: jsonData
+                                                                          options: 0
+                                                                            error: & errorParsingSurveyContent];
+            if (! surveyContent)
+            {
+                NSError *error = [NSError errorWithCode: APCErrorParsingSurveyContentCode
+                                                 domain: APCErrorDomain
+                                          failureReason: APCErrorParsingSurveyContentReason
+                                     recoverySuggestion: APCErrorParsingSurveyContentSuggestion
+                                            nestedError: errorParsingSurveyContent];
+                
+                APCLogError2 (error);
+            }
+            
+            else
+            {
+                @try
+                {
+                    id manager = SBBComponent(SBBSurveyManager);
+                    SBBSurvey *survey = [[manager objectManager] objectFromBridgeJSON: surveyContent];
+                    rkSurvey = [APCTask rkTaskFromSBBSurvey: survey];
+                }
+                @catch (NSException *exception)
+                {
+                    NSError *error = [NSError errorWithCode: APCErrorLoadingNativeBridgeSurveyObjectCode
+                                                     domain: APCErrorDomain
+                                              failureReason: APCErrorLoadingNativeBridgeSurveyObjectReason
+                                         recoverySuggestion: APCErrorLoadingNativeBridgeSurveyObjectSuggestion
+                                            relatedFilePath: surveyFilePath
+                                                 relatedURL: nil
+                                                nestedError: nil
+                                              otherUserInfo: @{ @"exception": exception,
+                                                                @"stackTrace": exception.callStackSymbols }];
+                    
+                    APCLogError2 (error);
+                }
+                @finally
+                {
+                    
+                }
+            }
+        }
+    }
+    
+    return rkSurvey;
+}
+
+// ---------------------------------------------------------
+#pragma mark - JSON conversions
+// ---------------------------------------------------------
+
+/**
+ Convert inbound Bridge server data to an NSDictionary of keys we know how to
+ look for.
+ 
+ This lets us use the same method to process data downloaded from the server as
+ we do data pulled from a local JSON file. It also lets us flatten out the
```

Comment:
I understand you're following the pattern of the schedule importer you're replacing, but does it even make sense to ship the app with tasks in a local JSON file, since tasks are scheduled for a specific day/time (as opposed to schedules where the app reads the JSON and generates the equivalent of tasks from the schedule therein, independent of when it's running)? And if not, should the code be left convoluted like this to support that use case? Would it make more sense to use the sageTask object more directly? (Or not? If there's other code elsewhere that would have to be refactored to accommodate that cleanup, maybe it's not worth it?)